### ISSUE TITLE
ewelink app requires location/GPS for wifi devices

### ISSUE THREAD

**Issue by jslay88** [itead/Sonoff_Devices_DIY_Tools#50]
The new update to the ewelink app has completely removed the concern for users privacy. In no way is it acceptable for me to have to give you access to my GPS on my device so that I can perform the quick pairing.

Because the DIY devices do not ship with a DIY compatible firmware, we are forced to do the initial FW upgrade through the ewelink app. This is no longer possible without giving access to GPS now when this has never been required before? The entire purpose for me purchasing these devices was for the DIY mode for privacy. You have now completely destroyed that capability

I either want this limitation removed from the app, or a refund for my devices, as I now have no way of getting them into DIY mode without flashing a 3rd party firmware via JTAG, which forces my warranty to be void. This limitation did not exist at the time of purchase, and has no reason to exist now.  

Between this and the virus being distributed here, my trust for Sonoff has gone to 0.

**Comment by than-sad** [comment]
I was losing two hours trying to pair my devices until i realized about the new location requirement.

I completely agree with you jslay88. I request this limitation removed from the app too.

**Comment by tomekit** [comment]
I've ordered few devices to play with them and choose the best solution to make my home smart.
I was really excited that these Sonoff devices have pretty good hardware for good price, unfortunately it's a huge no, so I am going to choose different DIY automation chip provider.

Sorry `itead`, but it's huge downside. I would prefer these devices to be more expensive than give out my privacy.

**Comment by tomekit** [comment]
I've got the temporary workaround which doesn't share your real location. It requires some effort and installing 3rd party app `FakeGPS`.

Here is the **trick** (on Android):
- Install FakeGPS app,
- In developer options set FakeGPS as the GPS mock,
- Using FakeGPS select map to choose the "fake" location,
- Give eWeLink location permission and pair device using quick connect,
- Revoke location permission given to eWeLink,
- Remove FakeGPS from developer options as the GPS mock to restore your phone location capabilities back to native.

**Comment by jslay88** [comment]
I have since flashed via JTAG and will not be purchasing anymore of these devices. Unsubscribing from this topic, as the manufacture clearly has checked out on this as well and doesn't care. 

Best of luck to the rest of you.

**Comment by scoulondre** [comment]
I agree 100% with the messages above. To be forced giving away our precise location, just in my case for bridging some wifi to 433mhz with Sonoff bridge far exceeds the technical requirements for such a function. There is clearly a will to record and use the maximum amount of personal data. I wonder if this is in accordance with GDPR regulations. However, I just returned my bridge today to Amazon. So you won't get my data neither my money.

**Comment by stpbrian** [comment]
I haven't tried the Fake GPS app solution, I'm currently more concerned with giving a Chinese company the wifi password to my home network.  Is anyone else concerned about this?  Is there any way to pair my device without giving them that info?

**Comment by elisey** [comment]
@stpbrian you can try to just update firmware on your device via eWeLink app, than put device into DIY mode. After that you can delete eWeLink app and turn off internet access for your device in firewall settings in your router. You still will be able to enter to the device settings via HTTP (http://10.10.7.1) and set your wifi credentials.

**Comment by Gelisob** [comment]
> @stpbrian you can try to just update firmware on your device via eWeLink app, than put device into DIY mode. After that you can delete eWeLink app and turn off internet access for your device in firewall settings in your router. You still will be able to enter to the device settings via HTTP (http://10.10.7.1) and set your wifi credentials.

That means you gave them your information, email, precise gps location. Does not matter that you delete the app afterwards. Thats the point, to pair your device without giving out personal info to Chinese company/coverment. 
Shipping the devices deliberately with not working DIY firmware (fails to connect) is quite telling.
Wish i knew before, also spent too much time on trying to simply get it to go into accessible DIY mode.
